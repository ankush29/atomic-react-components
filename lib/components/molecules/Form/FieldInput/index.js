// @flow

export { default } from './FieldInput';
