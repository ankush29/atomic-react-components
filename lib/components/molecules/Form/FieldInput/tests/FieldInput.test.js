import React from 'react';
import toJson from 'enzyme-to-json';
import { shallowWithTheme, renderWithTheme } from '../../../../../../__mock__/themeMock';
import { FieldInputVanilla } from '../FieldInput';

describe('FieldInput Component', () => {
  let FieldInput = '';

  test.only('should render correctly', () => {
    FieldInput = shallowWithTheme(<FieldInputVanilla />);
    expect(toJson(FieldInput)).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Enter username',
    };
    FieldInput = renderWithTheme(<FieldInputVanilla {...props} />);
    expect(FieldInput.find('label').text()).toEqual('Enter username');
  });

  test('should render Input component with proper type', () => {
    const props = {
      name: 'email',
      type: 'email',
      id: 'email',
    };
    FieldInput = renderWithTheme(<FieldInputVanilla {...props} />);
    expect(FieldInput.find('input[type=email]')).toHaveLength(1);
  });
});
