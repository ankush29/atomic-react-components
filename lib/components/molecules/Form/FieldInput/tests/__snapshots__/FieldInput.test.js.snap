// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput Component should render correctly 1`] = `
<div
  className=""
>
  <Label
    className=""
    elementLocator=""
    htmlFor="input-jrkyfm80-1"
  />
  <FormikConnect(FieldInner)
    id="input-jrkyfm80-1"
    render={[Function]}
    type="text"
  />
  <FormikConnect(ErrorMessageImpl)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "FieldError-sc-1b3h259-0",
          "isStatic": false,
          "rules": Array [
            "font-family:",
            [Function],
            ";font-size:12px;text-align:left;color:",
            [Function],
            ";margin:0;",
            [Function],
            ";",
            ";",
          ],
        },
        "displayName": "FieldError",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "FieldError-sc-1b3h259-0",
        "target": [Function],
        "toString": [Function],
        "usesTheme": true,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
  />
</div>
`;
